board/can_comms.h

```diff
@@ -68,8 +68,9 @@ void comms_can_write(const uint8_t *data, uint32_t len) {
       can_write_buffer.ptr += can_write_buffer.tail_size;
       pos += can_write_buffer.tail_size;
 
+      void *addr = &to_push;
```

Comment:
this looks like a workaround, rather than a real fix. it's also possible we may need to suppress something for memcpy

Reply:
It is indeed a workaround. This fixes the error that happens because we pass an uninitialized variable to a function. My justification for this change was that this was an oddity rather than a regular occurrence for the memcpy function. If you believe that we may need to call uninitialized variables from memcpy, we can suppress that check on the function.

[Reference](https://trac.cppcheck.net/ticket/1203)

Reply:
After further introspection and digging deeper into the cppchecker code and issues, this seems to be a false positive caused by the return statement in the memcpy function.
An [issue](https://trac.cppcheck.net/ticket/12414#no1) has already been created for this in the cppchecker's issue tracker.

We seem to have three options here:
- suppress the check for memcpy function
-  change the return statement for memcpy to a const value
- fix the false positive in cppchecker

@adeebshihadeh let me know how to move forward with this.

Reply:
hmm, seems like this is a regression in cppcheck and we should just stick to the releases. I propose we go back to the 2.13 release and just `git cherry-pick` your fix onto it